recipes/embree/all/conanfile.py

```diff
@@ -0,0 +1,107 @@
+    _source_folder = "source_subfolder"
+    _build_folder = "build_subfolder"
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "compiler", "build_type", "arch"
+    requires = "tbb/2019_u9"
+
+    options = {
+        "shared": [
+            True,
+            False
+        ],
+        "fPIC": [
+            True,
+            False
+        ],
+        "backface_culling": [
+            True,
+            False
+        ],
+        "ignore_invalid_rays": [
+            True,
+            False
+        ],
+        "ray_masking": [
+            True,
+            False
+        ]
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "backface_culling": False,
+        "ignore_invalid_rays": False,
+        "ray_masking": False
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        target_name = "{0}-{1}".format(self.name, self.version)
+        os.rename(target_name, self._source_folder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+
+        # Configure CMake library build:
+        cmake.definitions["EMBREE_STATIC_LIB"] = not self.options.shared
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.definitions["EMBREE_TUTORIALS"] = False
+        cmake.definitions["EMBREE_GEOMETRY_CURVE"] = True
+        cmake.definitions["EMBREE_GEOMETRY_GRID"] = True
+        cmake.definitions["EMBREE_GEOMETRY_INSTANCE"] = True
+        cmake.definitions["EMBREE_GEOMETRY_QUAD"] = True
+        cmake.definitions["EMBREE_GEOMETRY_SUBDIVISION"] = True
+        cmake.definitions["EMBREE_GEOMETRY_TRIANGLE"] = True
+        cmake.definitions["EMBREE_GEOMETRY_USER"] = True
+        cmake.definitions["EMBREE_RAY_PACKETS"] = True
+        cmake.definitions["EMBREE_RAY_MASK"] = self.options.ray_masking
+        cmake.definitions["EMBREE_BACKFACE_CULLING"] = self.options.backface_culling
+        cmake.definitions["EMBREE_IGNORE_INVALID_RAYS"] = self.options.ignore_invalid_rays
+        cmake.definitions["EMBREE_ISPC_SUPPORT"] = False
+
+        cmake.configure(build_folder=self._build_folder)
+        return cmake
+
+    def build(self):
+        os.remove(os.path.join(self._source_folder, "common", "cmake", "FindTBB.cmake"))
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE.txt", src=self._source_folder, dst="licenses")
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, 'lib', 'cmake'))
+
+    def package_info(self):
+        self.cpp_info.libdirs = [
+            "lib"
+        ]
+
+        self.cpp_info.includedirs = [
+            "include"
+        ]
+
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
libs order is important.
Seems like this order can be used (tried with several profiles, but I still have undefined symbols on linux with clang):
["embree3", "embree_sse42", "embree_avx", "embree_avx2", "embree_avx512knl", "embree_avx512skx", "simd", "lexers", "tasking", "sys", "math"]

Reply:
I changed to the list you suggested

Reply:
I changed to the list you suggested

Reply:
It seems like I cannot use it since embree_avx512knl will be missing.